pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/bucket/BucketDelayedDeliveryTracker.java

```diff
@@ -308,6 +375,10 @@ public synchronized NavigableSet<PositionImpl> getScheduledMessages(int maxMessa
                 try {
                     bucket.asyncLoadNextBucketSnapshotEntry().thenAccept(indexList -> {
                         if (CollectionUtils.isEmpty(indexList)) {
+                            synchronized (immutableBuckets) {
+                                immutableBuckets.remove(Range.closed(bucket.startLedgerId, bucket.endLedgerId));
+                            }
+                            bucket.asyncDeleteBucketSnapshot();
```

Comment:
We need a log to show the exception.

Reply:
Add exception log in `asyncDeleteBucketSnapshot`